Realm/Tests/RLMTestCase.m

```diff
@@ -149,15 +149,20 @@ - (RLMRealm *)realmWithTestPathAndSchema:(RLMSchema *)schema {
 
 - (void)waitForNotification:(NSString *)expectedNote realm:(RLMRealm *)realm block:(dispatch_block_t)block {
     XCTestExpectation *notificationFired = [self expectationWithDescription:@"notification fired"];
-    RLMNotificationToken *token = [realm addNotificationBlock:^(__unused NSString *note, RLMRealm *realm) {
+    RLMNotificationToken *token = [realm addNotificationBlock:^(NSString *note, RLMRealm *realm) {
+        XCTAssertNotNil(note, @"Note should not be nil");
         XCTAssertNotNil(realm, @"Realm should not be nil");
-        if (note == expectedNote) {
+        if ([note isEqualToString:expectedNote]) {
```

Comment:
This was deliberately checking pointer equality so that it would fail if the string constant for the notification is not used.